lib/pengine/native.c

```diff
@@ -567,7 +567,7 @@ pcmk__native_output_string(const pe_resource_t *rsc, const char *name,
     if ((node == NULL) && (rsc->lock_node != NULL)) {
         node = rsc->lock_node;
     }
-    if (pcmk_is_set(show_opts, pcmk_show_rsc_only)
+    if (pcmk_any_flags_set(show_opts, pcmk_show_rsc_only|pcmk_show_description)
         || pcmk__list_of_multiple(rsc->running_on)) {
```

Comment:
This is where the node is getting stripped out.  The block here sets node to NULL, and then just a little farther down we add the node but only if it's not NULL.  I'm not exactly sure, but I think you probably just want to remove the check on pcmk_show_description here.